fs/filter/filter.go

```diff
@@ -454,10 +454,17 @@ func (f *Filter) MakeListR(ctx context.Context, NewObject func(ctx context.Conte
 				return nil
 			})
 		}
-		for remote := range f.files {
-			remotes <- remote
-		}
-		close(remotes)
+		go func() {
```

Comment:
I think this can be written without the extra go routine which I think makes the logic a bit simpler

```go
	outer:
		for remote := range f.files {
			select {
			case remotes <- remote:
			case <-gCtx.Done():
				break outer
			}
		}
		close(remotes)
```

Reply:
Thanks for the suggestion, it was very helpful